### PR TITLE
feat: support Bun package manager

### PR DESCRIPTION
**Pull request by colinhacks**
Support Bun package manager.

Related: 
- https://github.com/tauri-apps/create-tauri-app/pull/481
- https://github.com/tauri-apps/tauri-docs/pull/1441

### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/cli/node/tauri.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 const cli = require('./main')
 const path = require('path')
 
-const [bin, script, ...arguments] = process.argv
```
Comment (colinhacks) [inline tooling/cli/node/tauri.js:10]: `arguments` is a reserved keyword in JavaScript and cannot be used as a variable name in strict mode: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments

File: tooling/cli/node/tauri.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ if (bin === '@tauri-apps/cli') {
 }
 // Even if started by a package manager, the binary will be NodeJS.
 // Some distribution still use "nodejs" as the binary name.
-else if (binStem.match(/(nodejs|node)\-?([0-9]*)*$/g)) {
+else if (binStem.match(/(nodejs|node|bun)\-?([0-9]*)*$/g)) {
```
Comment (colinhacks) [inline tooling/cli/node/tauri.js:23]: Running a script with `bunx --bun` will execute it with the `bun` executable even if the shebang says `node`

File: tooling/cli/node/index.d.ts [hunk 3]
```diff
@@ -1,7 +1,3 @@
-// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
-// SPDX-License-Identifier: Apache-2.0
-// SPDX-License-Identifier: MIT
-
```
Comment (amrbashir) [inline tooling/cli/node/index.d.ts]: Let's revert these changes

File: tooling/cli/node/index.js [hunk 4]
```diff
@@ -1,7 +1,3 @@
-// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
-// SPDX-License-Identifier: Apache-2.0
-// SPDX-License-Identifier: MIT
-
```
Comment (amrbashir) [inline tooling/cli/node/index.js]: ditto

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks for your contribution, could you please rebase and target the 1.x branch and run `cargo fmt`? Don't worry about the `clippy` lints, I can fix them later.

**Comment by amrbashir** [comment]
I should've caught this sooner but looks like you didn't sign your commits and I can't merge the PR unless they are signed. You need to setup [commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits), then you can sign past commit like [this](https://superuser.com/questions/397149/can-you-gpg-sign-old-commits/1123928#1123928) for example.

**Comment by colinhacks** [comment]
Rebase was a mess so here's a new PR: https://github.com/tauri-apps/tauri/pull/7723/files
